packages/grafana-ui/src/components/FilterPill/FilterPill.tsx

```diff
@@ -7,23 +7,48 @@ import { useStyles2 } from '../../themes';
 import { IconName } from '../../types';
 import { clearButtonStyles } from '../Button';
 import { Icon } from '../Icon/Icon';
+import { Tooltip, TooltipPlacement } from '../Tooltip';
```

Comment:
@L-M-K-B @joshhunt (not sure who to ping here). I did some additions here to make the pill more usable. If that is not desirable I can just make my own but to me they seem sensible. ClassName I think should be standard prop but also I wanted to override the `display: flex` to `display: inline-flex` which imho makes more sense for this cause I assume I would mostly just use the in a line not in a block context.

Reply:
Adding className to this component will only ever make it _less_ reliable and more likely to break as we cannot take into consideration everyone's custom css whenever changes to this component are made.

How is this component managed to be used elsewhere without this change? 

I'm generally supportive of just changing this component to `display: inline-flex`, but I'm unsure how much of a breaking change this would be...

Reply:
Instead, you should be able to use Stack

```ts
<Stack gap={2}>
  <FilterPill ... />
  <Icon name="arrow-right"/> 
  <FilterPill ... />
  <Icon name="arrow-right"/> 
  <FilterPill ... />
</Stack>
```

(Note the use of an actual chevron icon instead of a greater-than symbol 😉 )

Reply:
> Adding className to this component will only ever make it less reliable and more likely to break

I am not sure I follow here. For example I would say margins should never be added to a library component. At the same time you need to define margins when using them in a layout and I feel like it makes more sense to add them to the component via `className` then to use a wrapper divs everytime. So I think `className` should be a standard props for this on every component in grafana/ui

Reply:
Ok I removed the FilterPill use here. Probably does not make sense anyway, the styling is just 4 lines of css so I may have been abusing that component a bit and don't want to do controversial changes in unrelated PR :D. I still think though that the className and not having styles that affect outsides of the component make sense for the library components.